utils/misc/retry.go

```diff
@@ -31,3 +31,29 @@ func RetryWith(parentContext context.Context, timeout time.Duration, maxAttempts
 	}
 	return err
 }
+
+func QueryWithRetries(parentContext context.Context, timeout time.Duration, maxAttempts int, f func(ctx context.Context) (interface{}, error)) (interface{}, error) {
```

Comment:
[minor] If this retry logic is specific to jobsdb I would suggest moving it there instead of the misc package.

https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

Reply:
yes, it is not being used anywhere as of now. But, the reasoning behind adding it in `misc` is, since this is a generic thing that we might want to do from multiple other packages. So, having it in `misc` gives visibility in future. And, helps in avoiding redundant code rewriting.

Reply:
After discussion, I understand that this function, with some improvements, can be a much better retry implementation of what the backoff library currently offers.